src/datasets/iterable_dataset.py

```diff
@@ -317,16 +317,40 @@ def __iter__(self):
 
     def _iter_arrow(self):
         shard_idx_start = self._state_dict["shard_idx"] if self._state_dict else 0
-        for gen_kwags in islice(_split_gen_kwargs(self.kwargs, max_num_jobs=self.num_shards), shard_idx_start, None):
+        kwargs_with_shuffled_shards = (
+            _shuffle_gen_kwargs(self.generator, self.kwargs) if hasattr(self, "generator") else self.kwargs
```

Comment:
is this expected ? I think ShuffledDataSourcesArrowExamplesIterable has its own `_iter_arrow()` implementation